arch/arm/armv7-m/src/arch_main.c

```diff
@@ -54,7 +54,19 @@ static void arch_init_ccr(void)
 
 int main(void)
 {
+    int status;
+
     arch_init_ccr();
 
-    return fwk_arch_init(&arch_init_driver);
+    status = fwk_arch_init(&arch_init_driver);
```

Comment:
see https://github.com/ARM-software/SCP-firmware/pull/524#issuecomment-934602620)

Reply:
Yes, I'm going to look more closely on your proposal and integrate it in the next version

Reply:
I have  studied your proposal and integrated it in the following way:
- armv7-m and armv-a are not modified by this PR anymore

- I have renamed SCP_ARCHITECTURE_VM into SCP_ARCHITECTURE_OS and BUILD_HAS_ARCH_VM into BUILD_HAS_ARCH_OS because zephyr and optee are not VM (but our main use case with zephyr is to run it into a VM)

- I have kept __fwk_thread_run calling __fwk_run_event in order to avoid duplicating the same code twice. The end result is the same for __fwk_thread_run() which is : 
  1- loop on ctx.event_queue and process event until the list is empty
  2- process_isr and continue to step 1 if there is an isr event
  3- unbuffer log 
  4- restart on 1

I have run few more test and will push the new version by end of the day